Bug: a directory with name *.yml makes yamllint to lint all containing files despite their type

Give the following tree structure:
/test.yml
/generated/products.yml/products.sql

When executing yamllint, it scans the file products.sql, which is not yaml, raising an error.

Run into this issue today as well:

```
 yamllint -s -c .yamllint .
./.ivy2/cache/org.yaml/snakeyaml/ivydata-1.17.properties
  1:2       warning  missing starting space in comment  (comments)
  2:2       warning  missing starting space in comment  (comments)
  3:1       warning  missing document start "---"  (document-start)

./.ivy2/cache/org.yaml/snakeyaml/ivy-1.17.xml.original
  1:1       warning  missing document start "---"  (document-start)
  1:121     error    line too long (201 > 120 characters)  (line-length)
  575:121   error    line too long (253 > 120 characters)  (line-length)
  578:121   error    line too long (436 > 120 characters)  (line-length)
  580:121   error    line too long (123 > 120 characters)  (line-length)
  582:121   error    line too long (136 > 120 characters)  (line-length)
```

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -73,6 +73,9 @@ class CommandLineTestCase(unittest.TestCase):
             # file in dir
             'sub/ok.yaml': '---\n'
                            'key: value\n',
+            # directory that looks like a yaml file
+            'sub/directory.yaml/not-yaml.txt': '',
+            'sub/directory.yaml/empty.yml': '',
             # file in very nested dir
             's/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml': '---\n'
                                                        'key: value\n'
@@ -108,6 +111,7 @@ class CommandLineTestCase(unittest.TestCase):
              os.path.join(self.wd, 'dos.yml'),
              os.path.join(self.wd, 'empty.yml'),
              os.path.join(self.wd, 's/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml'),
+             os.path.join(self.wd, 'sub/directory.yaml/empty.yml'),
              os.path.join(self.wd, 'sub/ok.yaml'),
              os.path.join(self.wd, 'warn.yaml')],
         )
@@ -132,6 +136,7 @@ class CommandLineTestCase(unittest.TestCase):
         self.assertEqual(
             sorted(cli.find_files_recursively(items, conf)),
             [os.path.join(self.wd, '/etc/another/file'),
+             os.path.join(self.wd, 'sub/directory.yaml/empty.yml'),
              os.path.join(self.wd, 'sub/ok.yaml')],
         )
 
@@ -152,7 +157,8 @@ class CommandLineTestCase(unittest.TestCase):
         self.assertEqual(
             sorted(cli.find_files_recursively([self.wd], conf)),
             [os.path.join(self.wd, 'dos.yml'),
-             os.path.join(self.wd, 'empty.yml')]
+             os.path.join(self.wd, 'empty.yml'),
+             os.path.join(self.wd, 'sub/directory.yaml/empty.yml')]
         )
 
         conf = config.YamlLintConfig('extends: default\n'
@@ -174,6 +180,8 @@ class CommandLineTestCase(unittest.TestCase):
              os.path.join(self.wd, 'no-yaml.json'),
              os.path.join(self.wd, 'non-ascii/éçäγλνπ¥/utf-8'),
              os.path.join(self.wd, 's/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml'),
+             os.path.join(self.wd, 'sub/directory.yaml/empty.yml'),
+             os.path.join(self.wd, 'sub/directory.yaml/not-yaml.txt'),
              os.path.join(self.wd, 'sub/ok.yaml'),
              os.path.join(self.wd, 'warn.yaml')]
         )
@@ -191,6 +199,8 @@ class CommandLineTestCase(unittest.TestCase):
              os.path.join(self.wd, 'no-yaml.json'),
              os.path.join(self.wd, 'non-ascii/éçäγλνπ¥/utf-8'),
              os.path.join(self.wd, 's/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml'),
+             os.path.join(self.wd, 'sub/directory.yaml/empty.yml'),
+             os.path.join(self.wd, 'sub/directory.yaml/not-yaml.txt'),
              os.path.join(self.wd, 'sub/ok.yaml'),
              os.path.join(self.wd, 'warn.yaml')]
         )

--- a/yamllint/config.py
+++ b/yamllint/config.py
@@ -46,7 +46,7 @@ class YamlLintConfig(object):
         return self.ignore and self.ignore.match_file(filepath)
 
     def is_yaml_file(self, filepath):
-        return self.yaml_files.match_file(filepath)
+        return self.yaml_files.match_file(os.path.basename(filepath))
 
     def enabled_rules(self, filepath):
         return [yamllint.rules.get(id) for id, val in self.rules.items()

